vortex-alp/src/compress.rs

```diff
@@ -50,41 +45,45 @@ impl EncodingCompression for ALPEncoding {
 
     fn compress(
         &self,
-        array: &dyn Array,
-        like: Option<&dyn Array>,
+        array: &Array,
+        like: Option<&Array>,
         ctx: CompressCtx,
-    ) -> VortexResult<ArrayRef> {
-        let like_alp = like.map(|like_array| like_array.as_alp());
+    ) -> VortexResult<Array<'static>> {
+        let like_alp = like.map(|like_array| like_array.as_array_ref());
+        let exponents_cloned = like
+            .map(|like_array| ALPArray::try_from(like_array).unwrap())
+            .as_ref()
```

Comment:
Why as_ref?

Reply:
vestigial from wrestling with the compiler, will remove